### ISSUE TITLE
Is the probability extracted corresponding to the selected token?

### ISSUE THREAD

**Issue by luke217** [lingchen0331/UQ_ICL#3]
Hi UQ_ICL authors!

According to the paper, the second step of the framework of entropy-based uncertainty estimation is "2) selecting token(s) that is relevant to the answer and extract the probabilities". However, I saw that in code, after applying softmax() on logits, you used [max()](https://github.com/lingchen0331/UQ_ICL/blob/main/sources/utils.py#L191) to pick the highest probability. Is it truly the probability associated with the selected token? 

Thanks!

**Comment by lingchen0331** [comment]
Hi, you can check this line: https://github.com/lingchen0331/UQ_ICL/blob/707e4df90a8d00b34c68ec9cf674675e8a19365b/sources/utils.py#L147C1-L148C1
It gets the index of the first token that corresponds to a number
